src/troute-network/troute/HYFeaturesNetwork.py

```diff
@@ -511,16 +512,19 @@ def build_qlateral_array(self, run,):
 
             dfs=[]
             for f in qlat_files:
-                df = read_file(f).set_index(['feature_id']) 
+                df = read_file(f)
+                df['feature_id'] = df['feature_id'].map(lambda x: int(str(x).removeprefix('nex-')) if str(x).startswith('nex') else int(x))
+                df = df.set_index('feature_id')
                 dfs.append(df)
             
             # lateral flows [m^3/s] are stored at NEXUS points with NEXUS ids
-            nexuses_lateralflows_df = pd.concat(dfs, axis=1)  
+            nexuses_lateralflows_df = pd.concat(dfs, axis=1) 
             
             # Take flowpath ids entering NEXUS and replace NEXUS ids by the upstream flowpath ids
-            qlats_df = pd.concat( (nexuses_lateralflows_df.loc[int(k)].rename(v)
-                                for k,v in self.downstream_flowpath_dict.items() ), axis=1
-                                ).T 
+            qlats_df = pd.concat(
+                (nexuses_lateralflows_df.loc[int(k)].rename(v) for k,v in self.downstream_flowpath_dict.items() if k in nexuses_lateralflows_df.index.to_list()),
+                axis=1
+            ).T 
```

Comment:
this seems to be entering an infinite loop when I run t-route with *CHRTOUT files as forcing. If you change this to:
`qlats_df = nexuses_lateralflows_df.rename(index=self.downstream_flowpath_dict)`
it works with *CHRTOUT files and still passes your pytest file.

Reply:
Awesome, I think this works well. Implemented in ebfdd6d.